app/actions/views/channel.js

```diff
@@ -183,6 +184,15 @@ export function handleSelectChannel(channelId) {
         const member = myMembers[channelId];
 
         if (channel) {
+            if (!member) {
```

Comment:
This logic should not be here

Reply:
Could you expand on your comment? From my testing it appears that when a user clicks on a link to a channel that they dont belong to, the mobile app tries to load the posts of the channel without being a member of the channel which leads to errors. 

Following is the server output from when this happens:

```
EDEBU[2020-09-16T18:56:47.6347246+05:30] Received HTTP request                         caller="web/handlers.go:100" method=GET request_id=95s8ujc7i7gx5bh1fbommypd5o status_code=200 url=/api/v4/teams/name/ad-1/channels/name/ruin-1
ERRO[2020-09-16T18:56:48.2814496+05:30] You do not have the appropriate permissions.  caller="mlog/log.go:229" err_details="userId=xboqzk3cfp8cbbujhzedh7rjmy, permission=read_channel," err_where=Permissions http_code=403 ip_addr=192.168.0.111 method=GET path=/api/v4/channels/jgdwn6o1ij8exdy8aofwb8xkne/posts request_id=fnfd8qshntn77ptj84bw6zc6cr user_id=xboqzk3cfp8cbbujhzedh7rjmy
DEBU[2020-09-16T18:56:48.2815175+05:30] Received HTTP request                         caller="web/handlers.go:100" method=GET request_id=fnfd8qshntn77ptj84bw6zc6cr status_code=403 url=/api/v4/channels/jgdwn6o1ij8exdy8aofwb8xkne/posts
WARN[2020-09-16T18:56:48.283442+05:30] Failed to get membership                      caller="app/channel.go:2413" error="resource: ChannelMember id: channelId=jgdwn6o1ij8exdy8aofwb8xkne, userId=xboqzk3cfp8cbbujhzedh7rjmy"
```

Reply:
`handleSelectChannel` is called in various places and for all channel types. That said adding the user to the channel should happen before calling this function as seen here https://github.com/mattermost/mattermost-mobile/blob/master/app/components/sidebars/main/main_sidebar_base.js#L109

In this case I would add the logic in `handleSelectChannelByName` and only attempt to join the channel if it is in fact a public channel only.